### PR TITLE
Generates kotlin base classes for particles reading tuples.

### PR DESCRIPTION
**Pull request by piotrswigon**
This unblocks having the right handle interfaces, but the underlying handles are not yet specified correctly. The follow up work is tracked by b/157598151 and will come as a next PR, as this PR is large enough as is.

### CODE REVIEW

**Comment by mykmartin** [comment]
... although I think the second part of `createNodes` is no longer correct (the bit that starts with "Recurse on any nested schemas").

IIUC you no longer need `createNodes` to recurse itself, because you'll now find the nested ref schemas in `collectSchemas`, but I think you still need to set up the `refs` inner graph structure.

File: javatests/arcs/schematests/joins/Join.arcs [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+  Reader
+    data: triples
+
+particle Reader in 'arcs.schematests.joins.Reader'
+  data: reads [(
+    &Product {name: Text, photo: URL},
+    &Review {author: Text, content: Text, rating: Number},
+    &Manufacturer {name: Text, address: Text}
+  )]
```
[the remark this answers is not in the document]
Comment (piotrswigon) [reply javatests/arcs/schematests/joins/Join.arcs]: Done

**Comment by piotrswigon** [comment]
Thank you both!

@mykmartin 

actually I think everything is still correct. `collectSchemas` will only return top-level schemas (i.e. one per connection unless tuples are involved). I will rename it to `topLevelSchemas`to convey this better.

@csilvestrini 

I didn't make it executable. It was executable, now I removed the executable bit, maybe it will help for the follow up PRs.

**Comment by mykmartin** [comment]
Oh, right. `collectSchemas` only "follows" contained types, it doesn't look at schema fields. All good.

**Pull request merged**